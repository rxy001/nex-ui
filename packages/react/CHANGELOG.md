# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
