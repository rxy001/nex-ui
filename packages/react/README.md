<div style="background:white;">
  <p align="center">
    <img src='../../assets/logo.png' style="width: 150px; height: 150px;" />
  </p>
  <h1 align="center">Nex UI</h1>
  <p align="center">ðŸŽ‰ A beautiful, modern, and reliable React component library.</p>
  <p align="center"><b>Theme Customizable, Light & Dark UI, Unique DX, Written In TypeScript</b></p>
</div>

## Documentation

Visit https://nex-ui-docs.vercel.app to view the full documentation.

## Getting Started

Visit https://nex-ui-docs.vercel.app/docs/getting-started to get started with Nex UI.

## Storybook

Visit https://nex-ui-storybook.vercel.app to view the storybook for all components.

## Contributing

Contributions are always welcome!

See [CONTRIBUTING.md](/CONTRIBUTING.md) for ways to get started.

Please adhere to this project's [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md).

## License

[MIT](https://choosealicense.com/licenses/mit/)
