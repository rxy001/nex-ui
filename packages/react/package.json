{
  "name": "@nex-ui/react",
  "version": "0.0.12",
  "author": "X1ng Yu",
  "description": "ðŸŽ‰ A beautiful, modern, and reliable React component library.",
  "homepage": "https://nex-ui-docs.vercel.app/",
  "private": false,
  "keywords": [
    "react",
    "react-component",
    "nui",
    "nex-ui"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rxy001/nex-ui.git",
    "directory": "packages/react"
  },
  "license": "MIT",
  "scripts": {
    "build": "tsx ../../scripts/build.ts --dts",
    "build:fast": "tsx ../../scripts/build.ts",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "src/index.ts",
  "devDependencies": {
    "@storybook/react": "8.6.12"
  },
  "dependencies": {
    "@nex-ui/hooks": "workspace:*",
    "@nex-ui/icons": "workspace:*",
    "@nex-ui/styled": "workspace:*",
    "@nex-ui/system": "workspace:*",
    "@nex-ui/utils": "workspace:*",
    "clsx": "2.1.1"
  },
  "peerDependencies": {
    "@emotion/react": ">=11",
    "motion": "^12.7.2",
    "react": ">=19",
    "react-dom": ">=19"
  },
  "sideEffects": false,
  "clean-package": "../../clean-package.config.json"
}
