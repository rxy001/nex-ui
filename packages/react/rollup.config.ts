import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { defineConfig } from 'rollup'
import typescript from '@rollup/plugin-typescript'
import del from 'rollup-plugin-delete'
import dts from 'rollup-plugin-dts'
import { nexUIRollupPlugin } from '@nex-ui/plugins'

const dirname = fileURLToPath(new URL('./', import.meta.url))

export default defineConfig([
  {
    plugins: [
      del({
        targets: './dist/*',
      }),
      typescript({
        tsconfig: path.resolve(dirname, 'tsconfig.json'),
      }),
      nexUIRollupPlugin(),
    ],
    input: './src/index.ts',
    external: [
      'react',
      'react-dom',
      'react/jsx-runtime',
      'react-dom/client',
      '@nex-ui/utils',
      '@nex-ui/css-system',
      'classnames',
      'framer-motion',
      '@iconify/react',
      '@vanilla-extract/dynamic',
      '@vanilla-extract/css',
      '@vanilla-extract/css/functionSerializer',
    ],
    output: [
      {
        preserveModules: true,
        format: 'es',
        dir: './dist/es',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
      {
        preserveModules: true,
        format: 'cjs',
        dir: './dist/lib',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
    ],
  },
  {
    plugins: [dts()],
    input: './src/index.ts',
    output: {
      dir: './dist/types',
      format: 'es',
      preserveModules: true,
    },
  },
])
