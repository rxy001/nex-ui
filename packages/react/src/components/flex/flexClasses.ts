import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const flexClasses = generateUtilityClasses('nui-flex', [
  // ---------root---------
  'root',
  'justify-center',
  'justify-start',
  'justify-end',
  'justify-flex-start',
  'justify-flex-end',
  'justify-left',
  'justify-right',
  'justify-space-between',
  'justify-space-around',
  'justify-space-evenly',
  'justify-revert',
  'justify-stretch',
  'justify-normal',
  'justify-revert-layer',

  'align-flex-start',
  'align-flex-end',
  'align-start',
  'align-end',
  'align-center',
  'align-self-start',
  'align-self-end',
  'align-baseline',
  'align-revert',
  'align-revert-layer',
  'align-normal',
  'align-stretch',

  'direction-column',
  'direction-column-reverse',
  'direction-row',
  'direction-row-reverse',
  'direction-revert',
  'direction-revert-layer',

  'wrap-nowrap',
  'wrap-wrap',
  'wrap-wrap-reverse',
  'wrap-revert',
  'wrap-revert-layer',

  'gap-10px',
  'gap-20px',

  'inline',

  // ---------root---------
])
