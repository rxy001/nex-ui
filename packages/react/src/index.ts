import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
