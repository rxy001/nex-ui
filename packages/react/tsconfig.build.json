{
  "extends": "../../tsconfig.json",
  "include": ["./src"],
  "exclude": ["./src/**/__tests__/**/*", "./src/**/__stories__/**/*"]
}
