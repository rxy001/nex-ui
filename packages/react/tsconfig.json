{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["../../*"],
      "@theme": ["./src/theme"]
    }
  },
  "include": ["./src", "./rollup.config.ts"]
}
