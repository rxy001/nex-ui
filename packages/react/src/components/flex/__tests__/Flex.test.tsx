import { createRef } from 'react'
import {
  testComponentStability,
  renderWithNexUIProvider,
  testRootClassName,
} from '~/tests/shared'
import { Flex } from '../index'
import { flexClasses } from '../flexClasses'

describe('Flex', () => {
  testComponentStability(<Flex />)

  testRootClassName(<Flex className='test-class' />, 'test-class')

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<Flex />)

    const flex = container.firstElementChild

    expect(flex).toHaveClass(flexClasses.root)
    expect(container.firstElementChild).toMatchSnapshot()
  })

  it("should forward ref to Flex's root element", () => {
    const ref = createRef<HTMLDivElement>()
    const { container } = renderWithNexUIProvider(<Flex ref={ref} />)
    expect(container.firstElementChild).toBe(ref.current)
  })

  it('should render inline flex when inline prop is true', () => {
    const { container } = renderWithNexUIProvider(<Flex inline />)

    expect(container.firstElementChild).toHaveStyleRule(
      'display',
      'inline-flex',
    )
  })

  it('should render flex with gap applied', () => {
    const { container } = renderWithNexUIProvider(<Flex gap='16px' />)

    expect(container.firstElementChild).toHaveStyleRule('gap', '16px')
  })

  it('should render flex with flex-direction applied', () => {
    const { container } = renderWithNexUIProvider(<Flex direction='column' />)

    expect(container.firstElementChild).toHaveStyleRule(
      'flex-direction',
      'column',
    )
  })

  it('should render flex with justify-content applied', () => {
    const { container } = renderWithNexUIProvider(<Flex justify='center' />)

    expect(container.firstElementChild).toHaveStyleRule(
      'justify-content',
      'center',
    )
  })

  it('should render flex with align-items applied', () => {
    const { container } = renderWithNexUIProvider(<Flex align='center' />)

    expect(container.firstElementChild).toHaveStyleRule('align-items', 'center')
  })

  it('should render flex with flex-wrap applied', () => {
    const { container } = renderWithNexUIProvider(<Flex wrap='wrap' />)

    expect(container.firstElementChild).toHaveStyleRule('flex-wrap', 'wrap')
  })
})
