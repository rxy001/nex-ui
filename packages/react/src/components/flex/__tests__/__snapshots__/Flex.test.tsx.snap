// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should render with default props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  class="nui-flex-root emotion-0"
/>
`;
